src/sonic-yang-models/yang-models/sonic-mclag.yang

```diff
@@ -0,0 +1,171 @@
+                leaf peer_link {
+                    description  "MCLAG Domain peer link, data backup path link between MCLAG peers";
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                    }
+                }
+                leaf keepalive_interval {
+                    description  "keepalive interval in seconds for MCLAG session between peers";
+                    type uint16 {
+                        range "1..60" {
+                            error-message "MCLAG Domain keepalive interval out of range";
+                            error-app-tag keepalive_interval-invalid;
+                        }
+                    }
+                    default 1;
+                }
+                leaf session_timeout {
+                    description  "Session timeout in seconds for MCLAG session between peers";
+                    type uint16 {
+                        range "1..3600" {
+                            error-message "MCLAG Domain session timeout out of range";
+                            error-app-tag session_timeout-invalid;
+                        }
+                    }
+                    default 30;
+                }
+                must "(keepalive_interval * 3) <= session_timeout" {
+                        error-message "(keepalive interval * 3) <= session_timeout value";
+                        error-app-tag keepalive_interval-invalid;
+                }
+            }
+        }
+
+        container MCLAG_INTERFACE { 
+
+            list MCLAG_INTERFACE_LIST { 
+                description  "List of MCLAG interfaces";
+                key "domain_id if_name";
+
+                leaf domain_id {
+                    description  "List of MCLAG interfaces";
+                    type leafref {
+                        path "../../../MCLAG_DOMAIN/MCLAG_DOMAIN_LIST/domain_id";
+                    }
+                }
+
+                leaf if_name {
+                    description  "MCLAG interface name";
+                    type leafref {
+                        path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                    }
+                }
+                leaf if_type {
+                    description  "MCLAG interface type, placeholder field to create instance";
+                    type string;
+                }
+            }
+        }
+
+        container MCLAG_UNIQUE_IP {
+
+            list MCLAG_UNIQUE_IP_LIST {
+                description  "List of VLANs allowed to have unique ip addresses on MCLAG Peer nodes";
+                key "if_name";
+
+                must "count(../../MCLAG_DOMAIN/MCLAG_DOMAIN_LIST/domain_id) != 0" {
```

Comment:
Do we have test for this must condition?

Reply:
yes we do